### ISSUE TITLE
Image upload continues after uploading all of their data

### ISSUE THREAD

**Issue by hamidi2** [telegramdesktop/tdesktop#26266]
### Steps to reproduce

Just drag and drop some images on the Telegram Desktop. The image upload to a channel begins.

### Expected behaviour

After uploads gets completed, I see no sign on the images and the channel update be visible to others in the channel.

### Actual behaviour

Sometimes the behavior is like what expected. But sometimes, especially when lots of images are going to be uploaded, the tick sign on some images, and the 100% progress indicator in some others, remain on the images. It seems that for some reason, the upload can't get completed. I mean the x sign inside a complete circle by 100% progress indicator. I see that nothing has remained to be uploaded, but the process doesn't get completed. In such a condition, it's more wonderful that Telegram is still uploading something! What is it uploading? The images data again?
I tried about 10000 images at the same time and grouped them. It took about an hour of half an hour for all of them to get uploaded. But it's about 24 hours Telegram is still uploading something. This didn't occur once. I examine it many times, although not every time. I've a good link to Telegram servers (about 3.2 GB/s). The images seem that are stood still in their statuses. Some have been completed with so sing, some have the tick sign, and some has the 100% progress indicator sign. Most of them are not visible by other Telegram clients.
Once I tried just one image. It remained in the 100% progress indicator status. I got disappointed. I thought that I should cancel the upload and upload it again. So, did the work and the same appeared. I let Telegram working. After half an hour, when I returned to PC, I found that it has been uploaded successfully.
What Telegram is uploading during this time?! Is it uploading the data erroneously in a loop, again and again?!
I use MATE, and installed telegram-desktop by sudo apt install.

### Operating system

Ubuntu 22.04

### Version of Telegram Desktop

3.6.1

### Installation source

Other (unofficial) source

### Logs

```shell
I don't know where's the Telegram's log.
```

**Comment by Aokromes** [comment]
plz try downloading official static (non-flakpak, non-snap) version from https://desktop.telegram.org

**Comment by hamidi2** [comment]
yeah, i'm trying it. this is what i thought it too.

**Comment by hamidi2** [comment]
Unfortunately, the latest version (4.8.1) has also the problem. What can I do?

**Comment by hamidi2** [comment]
It seems that Telegram can't handle more than about 500 images (50 posts) at a time. Maybe a bit more, but I get reasonable result with this amount. If the value exceeds, it seems that something gets wrong with Telegram. I hope the developers get this feedback and think about it. It should not be difficult to store the dropped files paths in a dynamic array and begin to upload them one after another. If the path length be at most 1000 bytes, 10000 images need not more than 10 MB. It should be easy to handle. I can't justify his behavior.

**Comment by hamidi2** [comment]
No! Sometime, it can't even upload 200 images properly.
Is there a way to tell Telegram to upload images sequentially? I think this behavior occurs because it want to upload them at the same time. After completing upload of one image, if it starts to upload the next one, I think it may be a workaround.

**Comment by hamidi2** [comment]
Another guess is that I use proxy server to connect to Telegram servers. In the first pass, Telegram sends all images data. In the second pass, it tries to finalize them. Some of them may have lost the connection and can't get finalized. My government decreases speed and drops some packets and Telegram doesn't retry, maybe.

**Comment by hamidi2** [comment]
Alo Telegram Desktop developers!
Where're you to answer?

**Comment by hamidi2** [comment]
It seems that Telegram servers limit the volume or number of clients uploads in a period of time. Is it right?
I tried the same channel in the time of a client couldn't upload anymore with another client and I couldn't succeed.
Maybe because of their servers bandwidths and their storage volumes, they don't allow their servers to be rushed! :dagger: 
Is it the case?

**Comment by hamidi2** [comment]
By my experience, I think the policy is applied to downloads too.

**Comment by hamidi2** [comment]
I still can't figure out why has been so much upload rate. In another test, I find that it's just waiting with no upload rate.

**Comment by john-preston** [comment]
@hamidi2 I think you're just hitting a flood control from the server. You can check logs if there are some different network errors, log for the official build can be found at ~/.local/share/TelegramDesktop/log.txt

**Comment by hamidi2** [comment]
Please give me your Telegram id, so that I may send you the log. I don't know what is "hitting a flood control from the server".

**Comment by hamidi2** [comment]
For example, last night I sent about 830 images to a channel. After sending, the images remained mostly in the finalizing status and there were no transaction anymore. Like giving passports in a border and waiting for certificates, they were in waiting mode. I waited about 15 minutes with no change. So, I had to remain PC turned on. Today, I see that the finalization process is done. Unlike my first experience, this time, there were no huge upload in the finalization process, just waiting.
